lightning/src/ln/channelmanager.rs

```diff
@@ -1405,7 +1414,7 @@ impl<Signer: Sign, M: Deref, T: Deref, K: Deref, F: Deref, L: Deref> ChannelMana
 		}
 	}
 
-	fn force_close_channel_with_peer(&self, channel_id: &[u8; 32], peer_node_id: Option<&PublicKey>) -> Result<PublicKey, APIError> {
+	fn force_close_channel_with_peer(&self, channel_id: &[u8; 32], peer_node_id: Option<&PublicKey>, peer_msg: Option<&String>) -> Result<PublicKey, APIError> {
```

Comment:
Can you add a quick doc to this method that notes that "`peer_node_id` should be set when we receive a message from a peer, but not set when the user closes, which will be re-exposed as the `ChannelClosed` reason"

Reply:
Taking Matt suggestion and renaming as remote `peer_node_id`. Sounds to me more lightweight to avoid confusion than a new enum ?

Reply:
We could rename `peer_node_id` to `initiator`. I do think an enum is less error-prone than relying on people to leave it as `None` when it's holder-initiated, as well as more readable

Reply:
Would you be happy if `peer_node_id` and `peer_msg` were merged, val?

Reply:
> Would you be happy if peer_node_id and peer_msg were merged, val?

That works for me!